core/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -346,6 +346,86 @@ class ApplicationInfo(
     }
   }
 
+  /**
+   * This function is meant to clean up Delta log execs so that you could align
+   * SQL ids between CPU and GPU eventlogs. It attempts to remove any delta log
+   * only SQL ids. This includes reading checkpoints, delta_log json files,
+   * updating Delta state cache/table. There is also one special case for
+   * MergeIntoCommandEdge that I think showed up because some eventlogs might not
+   * be on the exact same data.
+   */
+  def cleanSQLIdsToAlign(): Seq[SQLCleanAndAlignIdsProfileResult] = {
+    val noDeltaLogs = sqlPlans.filter {
```

Comment:
nit: we have 2 helper objects for Delta. If we define a method that returns non-delta logs in SQL plans, then we will be able to reuse the code (including testing)

The two classes that already defined in the code set are:

- [DatabricksParseHelper](https://github.com/NVIDIA/spark-rapids-tools/blob/dev/core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/DatabricksParseHelper.scala)
- [DeltaLakeHelper](https://github.com/NVIDIA/spark-rapids-tools/blob/dev/core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/DeltaLakeHelper.scala)

One more thing we can consider: To use a n override of ToolPlanGraph that already does that functionality.